crates/noirc_frontend/src/hir/def_collector/dc_crate.rs

```diff
@@ -337,6 +347,29 @@ fn resolve_struct_fields(
     (generics, fields)
 }
 
+/// Create the mappings from TyAliasId -> TypeAliasType
+/// so that expressions can access the fields of structs
+fn resolve_type_aliases(
+    context: &mut Context,
+    type_aliases: HashMap<TyAliasId, UnresolvedTypeAlias>,
+    crate_id: CrateId,
+) {
+    // We must first go through the struct list once to ensure all IDs are pushed to
+    // the def_interner map. This lets structs refer to each other regardless of declaration order
+    // without resolve_struct_fields non-deterministically unwrapping a value
+    // that isn't in the HashMap.
+    for (type_id, unresolved_typ) in &type_aliases {
+        let path_resolver = StandardPathResolver::new(ModuleId {
+            local_id: unresolved_typ.module_id,
+            krate: crate_id,
+        });
+        let file = unresolved_typ.file_id;
+        let ty = Resolver::new(&mut context.def_interner, &path_resolver, &context.def_maps, file)
+            .resolve_type(unresolved_typ.type_alias_def.ty.clone());
```

Comment:
After you're done with a Resolver, you'll need to take any errors it had and add them to the errors list, which should be added as a parameter of this function